### PR TITLE
Documentation for DeepSpeed Accelerator Abstraction Interface

### PR DESCRIPTION
**Pull request by delock**
This PR add a document as a tutorial how to use DeepSpeed Accelerator Abstraction Interface to write accelerator agnostic DeepSpeed models; how to run a DeepSpeed model on different accelerator through different environment; how to write accelerator extension for new accelerator.

### CODE REVIEW

File: docs/_tutorials/accelerator-abstraction-interface.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+  * [Write accelerator agnostic models](#write-accelerator-agnostic-models)
+    * [Port accelerator runtime calls](#port-accelerator-runtime-calls)
+    * [Port accelerator device name](#port-accelerator-device-name)
+    * [Tensor operations](#tensor-operations)
+    * [Communication backend](#communication-backend)
+  * [Run DeepSpeed model on different accelerators](#run-deepspeed-model-on-different-accelerators)
+  * [Implement new accelerator extension](#implement-new-accelerator-extension)
+
+# Introduction
+The DeepSpeed Accelerator Abstraction allows user to run large language model seamlessly on various Deep Learning acceleration hardware seamlessly with DeepSpeed.   It offers a set of accelerator runtime and accelerator op builder interface which can be implemented for different hardware.  This means user can write large language model code without hardware specific code.  With DeepSpeed Accelerator Abstraction, the same large language model can run on different hardware platform, without the need to rewrite model code.  This makes running large language model on different hardware easier.
```
Comment (delock) [inline docs/_tutorials/accelerator-abstraction-interface.md:17]: remove one 'seamlessly' from this line.

**Comment by delock** [comment]
Hi @tjruwase can you review and see if this PR needs further change?  Hope this tutorial could be useful for people who want to write accelerator agonistic model or want to develop new accelerator for DeepSpeed.

**Comment by tjruwase** [comment]
> Hi @tjruwase can you review and see if this PR needs further change? Hope this tutorial could be useful for people who want to write accelerator agonistic model or want to develop new accelerator for DeepSpeed.

Thanks @delock. I will review asap. 

@mrwyattii, @jeffra, FYI

**Review by jeffra — APPROVED** [review]
Thank you @delock!

**Comment by jeffra** [comment]
Once this is live it might be nice to update the formatter to link to this page when cuda references are found. We've had some folks ask for more details on why/how to fix the errors the formatter raises and your new doc i think will add further context for them :)

**Pull request merged**